plenum/test/view_change/test_msgHasAcceptableViewNo.py

```diff
@@ -0,0 +1,53 @@
+
+from plenum.common.messages.node_messages import ViewChangeDone
+from plenum.server.node import Node
+
+
+def test_future_vcdone_vc_in_progress(fake_node):
+    """
+    If from_current_state is False, then message should be put only into msgsForFutureViews queue
+    """
+    frm = 'Node3'
+    fake_node.view_changer.view_change_in_progress = True
+    current_view = fake_node.viewNo
+    proposed_view_no = current_view + 1
+    msg = ViewChangeDone(proposed_view_no, frm, fake_node.ledger_summary)
+    res = Node.msgHasAcceptableViewNo(fake_node, msg, frm)
+    assert proposed_view_no in fake_node.msgsForFutureViews
+    assert proposed_view_no not in fake_node.msgsToViewChanger
+    assert not res
+
+
+def test_future_vcdone_vc_not_in_progress(fake_node):
```

Comment:
These two tests can be parametrized with params=['view_change_in_progress', 'view_change_not_in_progress'], can't they?